opt/scopeserver/gen_test_loom.py

```diff
@@ -0,0 +1,256 @@
+        "Embeddings_X": dfToNamedMatrix(embeddings_X),
+        "Embeddings_Y": dfToNamedMatrix(embeddings_Y),
+    }
+    row_attrs = {}
+
+    return metadata, col_attrs, row_attrs
+
+
+def generate_clusterings():
+    # Clusters
+    metadata = {"clusterings": []}
+    col_attrs = {}
+    row_attrs = {}
+
+    clusterings = pd.DataFrame()
+    clusterings["0"] = [int(n / (num_cells / 4)) for n in range(num_cells)]
+    clusterings["1"] = [n % 4 for n in range(num_cells)]
+    col_attrs["ClusterID"] = clusterings["0"].values
+
+    for n, clustering in enumerate(clusterings.columns):
+        cluster_meta = {
+            "id": n,
+            "group": "Test clusterings",
+            "name": f"Cluster set {n}",
+            "clusters": [],
+            "clusterMarkerMetrics": [
+                {"accessor": "Test_Metric", "name": "Test Metric", "description": "Some test values to display"}
+            ],
+        }
+        cluster_markers = pd.DataFrame(
+            index=gene_ids, columns=[str(x) for x in range(max(set([int(x) for x in clusterings[clustering]])) + 1)]
+        )
+        cluster_markers_test_metric = pd.DataFrame(
+            index=gene_ids, columns=[str(x) for x in range(max(set([int(x) for x in clusterings[clustering]])) + 1)]
+        )
+        cluster_markers.fillna(0, inplace=True)
+        cluster_markers_test_metric.fillna(0, inplace=True)
+        for cluster in range(4):
+            cluster_markers.iloc[cluster * 4 : (cluster + 1) * 4, cluster] = 1
+            cluster_markers_test_metric.iloc[cluster * 4 : (cluster + 1) * 4, cluster] = cluster * 1 * n
+
+            clustDict = {}
+            clustDict["id"] = cluster
+            clustDict["description"] = f"Unannotated Cluster {str(cluster+1)}"
+            cluster_meta["clusters"].append(clustDict)
+
+        row_attrs[f"ClusterMarkers_{n}"] = dfToNamedMatrix(cluster_markers)
+        row_attrs[f"ClusterMarkers_{n}_Test_Metric"] = dfToNamedMatrix(cluster_markers_test_metric)
+        metadata["clusterings"].append(cluster_meta)
+
+    col_attrs["Clusterings"] = dfToNamedMatrix(clusterings)
+    return metadata, col_attrs, row_attrs
+
+
+def generate_test_loom_data():
+    matrix = generate_matrix()
+    embeddings_meta, embeddings_cols, embeddings_rows = generate_embeddings()
+    clusterings_meta, clusterings_cols, clusterings_rows = generate_clusterings()
+    regulons_meta, regulons_cols, regulons_rows = generate_regulons()
+
+    col_attrs = {
+        "CellID": np.array(cell_ids),
+        "nUMI": np.array(matrix.sum(axis=0)),
+        "nGene": np.array((matrix > 0).sum(axis=0)),
+        "Half cells": ["First half" if n % 2 == 0 else "Second half" for n in range(num_cells)],
+        **embeddings_cols,
+        **clusterings_cols,
+        **regulons_cols,
+    }
+
+    row_attrs = {
+        "Gene": gene_ids,
+        **embeddings_rows,
+        **clusterings_rows,
+        **regulons_rows,
+    }
+
+    metaJson = {
```

Comment:
I would also suggest to rename it
```suggestion
    meta_json = {
```